### PR TITLE
Rehaul Svelte documentation to focus on vite-powered set-ups

### LINKED ISSUE #1072: Improve documentation for Svelte chart wrappers
Note to self on improving the Svelte docs:

- user should not need to install `@carbon/charts` because it is a direct dependency of `@carbon/charts-svelte`
- document usage using the most popular example set-ups
- document CSS usage (vendor vs. CDN)
- document event listeners
- instruct users to install Svelte wrappers as *development* dependencies
- prefer using shipped CSS instead of CDN styles
- ~~install `carbon-components` as a development dependency~~
- add notes on `vite`, `SvelteKit` usage (related #1024, #1070)
- add notes on `rollup` usage (related #860)
- dynamic, client-side theming? (#966)
- remove `carbon-components` as a peer dependency as it is vendored

### Example set-ups

- [x] rollup
- [x] sapper (SSR error "document is not defined", fixed by #1073)
- [x] snowpack
- [x] sveltekit (SSR error "document is not defined", fixed by #1073)
- [x] svite
- [x] webpack

### PR DESCRIPTION
**Pull request by metonym**
Closes #1072 

- [x] omit `@carbon/charts` in installation command because it is a direct dependency of `@carbon/charts-svelte`
- [x] instruct users to install Svelte wrappers as *development* dependencies
- [x] remove `carbon-components` from peerDependencies as it is vendored by `@carbon/charts`
- [x] document CSS usage to use shipped StyleSheet instead of a CDN
- [x] add example for adding event listeners
- [x] add example for dispatched events (`on:load`, `on:update`, `on:destroy`)
- [x] add note on TypeScript usage, link to `types/` folder
- [x] add dynamic import example
- [ ] ~~document dynamic, client-side theming?~~ (#966)
- [x] document usage using the most popular example set-ups
  - [x] SvelteKit
  - [x] vite
  - [x] Sapper
  - [x] Rollup
  - [x] Webpack
  - [x] Snowpack

### Updates

**Fixes**

- chore(deps): remove `carbon-components` from `peerDependencies`

**Documentation**

- docs: update Svelte charts installation/set-up instructions
- docs: update Svelte chart examples
- docs: add note on TypeScript usage

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/carbon-design-system/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/svelte/README.md [hunk 1]
```diff
@@ -7,124 +7,333 @@
...
-npm install -S @carbon/charts @carbon/charts-svelte d3@5.x
+npm install -D @carbon/charts-svelte d3@5.x
 ```
 
 If you prefer [Yarn](https://yarnpkg.com/en/), use the following command
 instead:
 
 ```bash
-yarn add @carbon/charts @carbon/charts-svelte d3@5.x
+yarn add -D @carbon/charts-svelte d3@5.x
```
Comment (theiliad) [inline packages/svelte/README.md:28]: are we sure? dev-dep?

File: packages/svelte/README.md [hunk 2]
```diff
@@ -7,124 +7,333 @@
...
+
+export default {
+	context: 'window',
+	// ...
+};
+```
+
+#### Troubleshooting
+
+##### "process is not defined"
```
Comment (theiliad) [inline packages/svelte/README.md:116]: this is probably irrelevant now since our rollup config does the same

**Review by theiliad — CHANGES_REQUESTED** [review]

File: packages/svelte/README.md (hunk 1 above)
Comment (metonym) [reply packages/svelte/README.md:28]: Yes. This is standard for installing Svelte components, particularly for Sapper.

File: packages/svelte/README.md (hunk 2 above)
Comment (metonym) [reply packages/svelte/README.md:116]: Unfortunately, I still encounter this error when using version ^0.41.90. Would you like me to record this as a separate issue?

Comment (theiliad) [reply packages/svelte/README.md:116]: Are you sure? I don't see any references to `process.` in https://unpkg.com/@carbon/charts@0.41.90/bundle.js

Comment (metonym) [reply packages/svelte/README.md:116]: Double checked. With a complete fresh install, the resolved deps are:

- @carbon/charts-svelte@^0.41.90
- @carbon/charts@^0.41.90
- carbon-components@10.5.0

The IIFE bundle generated by *Rollup* still pulls in the `warning.js` from `warning@3.0.0`:

Screen Shot 2021-07-21 at 12 57 21 PM

Comment (theiliad) [reply packages/svelte/README.md:116]: oh ok, that'd be related to your svelte app I think, the charts bundle doesn't include references to `process`. Where is this piece of code coming from?

Comment (metonym) [reply packages/svelte/README.md:116]: Exactly. It's coming from `/node_modules/warning/browser.js`

**Review by theiliad — APPROVED** [review]

**Pull request merged**